src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -186,7 +186,10 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
 
             logger.Info(ErrorCode.SiloGcSetting, "Silo starting with GC settings: ServerGC={0} GCLatencyMode={1}", GCSettings.IsServerGC, Enum.GetName(typeof(GCLatencyMode), GCSettings.LatencyMode));
             if (!GCSettings.IsServerGC || !GCSettings.LatencyMode.Equals(GCLatencyMode.Batch))
```

Comment:
> My concern is that the warning on a single-core machine isn't actionable
Yes it is - bump up the VM's number of cores :-)

The prominent scenario in which application developers get tripped-up by ServerGC configuration is when they try to set-up Orleans in Azure. They tend to be surprised that ServerGC does not work in their dev-test VMs (which tend to be the cheapest option possible, hence single-core)
So, instead of banging their heads against the wall as to why ServerGC does not work (and yes, the fact that ServerGC only works on multi-core system is not common knowledge), they get a hint.

Anyway, this was the intent. Feel free to close the PR if you find it not useful.